src/overseers/overseer.py

```diff
@@ -66,33 +88,55 @@ def mytardis_setup(self) -> IntrospectionConfig:
         """Getter for mytardis_setup. Sends API request if self._mytardis_setup is None"""
         return self._mytardis_setup or self.get_mytardis_setup()
 
-    @staticmethod
-    def resource_uri_to_id(uri: URI) -> int:
-        """Gets the id from a resource URI
+    def type_supports_identifiers(self, object_type: MyTardisObject) -> bool:
+        """Check if the given object type supports identifiers"""
+        if self.mytardis_setup.objects_with_ids is None:
+            return False
+        # pylint: disable=unsupported-membership-test
+        return object_type in self.mytardis_setup.objects_with_ids
 
-        Takes resource URI like: http://example.org/api/v1/experiment/998
-        and returns just the id value (998).
+    def generate_identifier_matchers(
+        self, object_type: MyTardisObject, object_data: dict[str, Any]
+    ) -> Generator[dict[str, Any], None, None]:
+        """Generate object matchers for identifiers"""
 
-        Args:
-            uri: str - the URI from MyTardis
+        if self.mytardis_setup.objects_with_ids is None:
+            raise ValueError("The MyTardis instance does not support identifier search")
 
-        Returns:
-            The integer id that maps to the URI
-        """
-        uri_sep: str = "/"
-        return int(urlparse(uri).path.rstrip(uri_sep).split(uri_sep).pop())
+        # pylint: disable=unsupported-membership-test
+        if object_type not in self.mytardis_setup.objects_with_ids:
```

Comment:
Is this a duplication of type_supports_identifiers() above?

Reply:
Yeah, you're right. I've now replaced `type_supports_identifiers()` with a method `check_identifiers_enabled_for_type()`, which basically asserts both that identifiers are enabled generally, and that the specific type has identifiers enabled, and tried to replace most of the checking with that, except in `generate_object_matchers()` where we check if an object has identifiers enabled, but it's not error if they aren't.